authors/apps/profiles/views.py

```diff
@@ -0,0 +1,74 @@
+from rest_framework.generics import ListAPIView, UpdateAPIView, RetrieveAPIView
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.exceptions import NotFound, PermissionDenied
+
+from .models import Profile
+from .serializers import UserProfileSerializer, UpdateUserProfileSerializer
+from .renderers import ProfileJSONRenderer
+
+class UserProfileView(RetrieveAPIView):
+    """ 
+    retrieves an authors profile information
+
+    """
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ProfileJSONRenderer,)
+    serializer_class = UserProfileSerializer
+
+    def get_queryset(self):
+        try:
+            profile = Profile.objects.select_related('user').get(
+                user__username=self.kwargs.get('username')
+            )
+            return profile
+        except Exception:
+
+            raise NotFound('User profile not found')
+
+    def retrieve(self, request, **kwargs):
+        data = self.get_queryset()
+        serializer = self.serializer_class(data, context={'request': request})
+
+        return Response(serializer.data)
+
+class AuthorsProfileListAPIView(ListAPIView):
+    """
+    Gets a list all authors profiles 
+    """
+
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ProfileJSONRenderer,)
+    serializer_class = UserProfileSerializer
+
+    #gets the whole list
+    def get_queryset(self):
+        #here  we filter 
+        #we filter so that the user doesnot see his profile in the list
+        queryset = Profile.objects.all().exclude(user=self.request.user)
+        return queryset
+
+class UpdateUserProfileView(UpdateAPIView):
+    #here we update the users profile
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ProfileJSONRenderer,)
+    serializer_class = UpdateUserProfileSerializer
+    queryset = Profile.objects.all()
+
+
+    #we make a query to see if the user who wants to make an update owns the account
+    def get_object(self):
+        queryset = self.filter_queryset(self.get_queryset())
+        obj = queryset.select_related('user').get(
```

Comment:
I have seen this block of code repeated severally, why don't you make it a function so it can be called anywhere it's needed. create a file called utilities and add the function there. then import it in the views file.

Reply:
This is the first statement
so this one queries kwargs from the Profile object to get the username
   ```
      user__username=self.kwargs.get('username')
   ```

This is the second statement
this other one queries the user object to get the username
   ```
      user__username=self.request.user.username
   ```
The variable names are the same but functionality is different